targets/csharp-unityv2/source/Shared/Internal/PlayFabHttp/PlayFabHTTP.cs

```diff
@@ -0,0 +1,337 @@
+            Application.runInBackground = true; // Http requests respond even if you lose focus
+#if !UNITY_WSA && !UNITY_WP8
+            if (PlayFabSettings.RequestType == WebRequestType.HttpWebRequest)
+                _internalHttp = new PlayFabWebRequest();
+#endif
+            if (_internalHttp == null)
+                _internalHttp = new PlayFabWWW();
+            _internalHttp.Awake();
+        }
+
+        /// <summary>
+        /// Internal method for Make API Calls
+        /// </summary>
+        /// <typeparam name="TRequestType"></typeparam>
+        /// <typeparam name="TResultType"></typeparam>
+        /// <param name="api"></param>
+        /// <param name="apiEndpoint"></param>
+        /// <param name="request"></param>
+        /// <param name="authType"></param>
+        /// <param name="resultCallback"></param>
+        /// <param name="errorCallback"></param>
+        /// <param name="customData"></param>
+        protected internal static void MakeApiCall<TRequestType, TResultType>(string api, string apiEndpoint,
+            TRequestType request, string authType, Action<TResultType> resultCallback,
+            Action<PlayFabError> errorCallback, object customData = null)
+        {
+            InitializeHttp();
+            SendEvent(request, null, ApiProcessingEventType.Pre);
+            _internalHttp.MakeApiCall(api, apiEndpoint, request, authType, resultCallback, errorCallback, customData);
+        }
+
+        /// <summary>
+        /// Monobehaviour Awake Method, gathers system data
+        /// </summary>
+        public override void Awake()
+        {
+            _gatherer.GatherData();
+            base.Awake();
+        }
+
+        /// <summary>
+        /// Monobehaviour OnEnable Method, registers logger
+        /// </summary>
+        public void OnEnable()
+        {
+            StartCoroutine(RegisterLogger());
+        }
+
+        /// <summary>
+        /// Monobehaviour OnDisable method, performs cleanup of Logger and bindings to Log Messages
+        /// </summary>
+        public void OnDisable()
+        {
+            if (!string.IsNullOrEmpty(PlayFabSettings.LoggerHost))
+            {
+                _logger = null;
+#if UNITY_5
+                UnityEngine.Application.logMessageReceivedThreaded -= HandleLogOutput;
+#else
+                UniytEngine.Application.RegisterLogCallback(null);
+#endif
+            }
+        }
+
+        /// <summary>
+        /// Monobehaviour Update method, logs System info data if gathered.
+        /// </summary>
+        public void Update()
+        {
+            _internalHttp.Update();
+            if (!_systemInfoLogged && PlayFabSettings.EnableRealTimeLogging)
+            {
+                _gatherer.EnqueueToLogger(_logMessageQueue);
+                _systemInfoLogged = true;
+            }
+        }
+
+        /// <summary>
+        /// Monobehaviour FixedUpdate, logs messages when enabled.
+        /// </summary>
+        void FixedUpdate()
```

Comment:
there is specific reason why..  we don't want to evaluate the Logging of messages on every Update tick, that is too often.  This mitigates the amount of times that we actually call to write the logs.   running the .Update() on the internalHttp is critical to do in the Update.  I have moved the enqueing code to the FixedUpdate.. as it should live there anyways.